trace_decoder/src/tries.rs

```diff
@@ -411,7 +498,101 @@ impl StateTrie for StateSmt {
             .map(|(addr, acct)| (keccak_hash::keccak(addr), *acct))
     }
     fn root(&self) -> H256 {
-        todo!()
+        conv_hash::smt2eth(self.as_smt().root)
+    }
+}
+
+impl StateSmt {
+    #[deprecated = "this should only be called from the frontend parsing"]
```

Comment:
Not sure I follow

Reply:
I'm trying to limit misuse of this API, but maybe this is too much